app/src/androidTest/java/org/simple/clinic/DatabaseMigrationAndroidTest.kt

```diff
@@ -2425,6 +2425,150 @@ class DatabaseMigrationAndroidTest {
+            'created_at', 'updated_at', NULL
+        )
+      """)
+    }
+
+    fun verifyMedicalHistory(
+        db: SupportSQLiteDatabase,
+        medicalHistoryUuid: UUID,
+        diagnosedWithHypertension: String,
+        isOnTreatmentForHypertension: String,
+        hasHadHeartAttack: String,
+        hasHadStroke: String,
+        hasHadKidneyDisease: String,
+        hasDiabetes: String
+    ) {
+      db.query("""
+        SELECT * FROM "MedicalHistory" WHERE "uuid" = '$medicalHistoryUuid'
+      """).use { cursor ->
+        cursor.moveToNext()
+        assertThat(cursor.string("diagnosedWithHypertension")).isEqualTo(diagnosedWithHypertension)
+        assertThat(cursor.string("isOnTreatmentForHypertension")).isEqualTo(isOnTreatmentForHypertension)
+        assertThat(cursor.string("hasHadHeartAttack")).isEqualTo(hasHadHeartAttack)
+        assertThat(cursor.string("hasHadStroke")).isEqualTo(hasHadStroke)
+        assertThat(cursor.string("hasHadKidneyDisease")).isEqualTo(hasHadKidneyDisease)
+        assertThat(cursor.string("hasDiabetes")).isEqualTo(hasDiabetes)
+      }
+    }
+
+    // given
+    val allUnknownAnswersUuid = UUID.fromString("b2e3b63f-9b09-4b8d-bfa7-af2ac5972786")
+    val allYesAnswersUuid = UUID.fromString("c11af491-d125-45af-9b3c-aa73510960c3")
+    val allNoAnswersUuid = UUID.fromString("05384716-65d7-4261-8bcf-f2faecf62bb2")
+    val mixedAnswersUuid = UUID.fromString("f0a50cf4-cf78-450a-979a-00df585fd642")
+
+    val db_v39 = helper.createDatabase(39)
+    saveMedicalHistory(
+        db = db_v39,
+        medicalHistoryUuid = allUnknownAnswersUuid,
+        diagnosedWithHypertension = "UNKNOWN",
+        isOnTreatmentForHypertension = "UNKNOWN",
+        hasHadHeartAttack = "UNKNOWN",
+        hasHadStroke = "UNKNOWN",
+        hasHadKidneyDisease = "UNKNOWN",
+        hasDiabetes = "UNKNOWN"
+    )
+    saveMedicalHistory(
+        db = db_v39,
+        medicalHistoryUuid = allYesAnswersUuid,
+        diagnosedWithHypertension = "YES",
+        isOnTreatmentForHypertension = "YES",
+        hasHadHeartAttack = "YES",
+        hasHadStroke = "YES",
+        hasHadKidneyDisease = "YES",
+        hasDiabetes = "YES"
+    )
+    saveMedicalHistory(
+        db = db_v39,
+        medicalHistoryUuid = allNoAnswersUuid,
+        diagnosedWithHypertension = "NO",
+        isOnTreatmentForHypertension = "NO",
+        hasHadHeartAttack = "NO",
+        hasHadStroke = "NO",
+        hasHadKidneyDisease = "NO",
+        hasDiabetes = "NO"
+    )
+    saveMedicalHistory(
+        db = db_v39,
+        medicalHistoryUuid = mixedAnswersUuid,
+        diagnosedWithHypertension = "YES",
+        isOnTreatmentForHypertension = "NO",
+        hasHadHeartAttack = "UNKNOWN",
+        hasHadStroke = "UNKNOWN",
+        hasHadKidneyDisease = "NO",
+        hasDiabetes = "YES"
+    )
+
+    // when
+    val db_v40 = helper.migrateTo(40)
+
+    // then
+    verifyMedicalHistory(
```

Comment:
It would be nice to combine the pre and post migration values and move them physically closer to each other. That will make the tests more readable (probably a Tuple7).

Reply:
Done. The arrow `Tuple7` was not a good fit since most of the value types are the same, so I introduced a data class scoped to the test to represent this.